doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+.. sectnum::
+
+#########################
+Amazon SageMaker Debugger
+#########################
+
+
+Amazon SageMaker Debugger allows you to detect anomalies while training your machine learning model by emitting relevant data during training, storing the data and thereafter analyzing it.
+
+.. contents::
+
+Background
+==========
+
+Amazon SageMaker provides every developer and data scientist with the ability to build, train, and deploy machine learning models quickly. Amazon SageMaker is a fully-managed service that encompasses the entire machine learning workflow. You can label and prepare your data, choose an algorithm, train a model, and then tune and optimize it for deployment. You can deploy your models to production with Amazon SageMaker to make predictions at lower costs than was previously possible.
+
+SageMaker Debugger provides a way to hook into the training process and emit debug artifacts (*aka* "tensors") that are representative of the state of the training at any given point in the training lifecycle. Debugger then stores the data in real time and provides a way for you to analyze the data through a concept called "rule". Rules encapsulate logic to analyze tensors and react to anomalies therein.
```

Comment:
* [@Vikas-kum] Provided a link to the sagemaker debugger main page. Not providing the link to the GitHub main page because that's the library for Debugger and isn't related to how to use Debugger in SageMaker through SageMaker Python SDK.
* [@aws-rbs] Thanks for the edit suggestion. Incorporated.